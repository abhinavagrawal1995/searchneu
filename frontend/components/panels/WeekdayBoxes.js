/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import css from './WeekdayBoxes.css';

const cx = classNames.bind(css);

// This file is responsible for the weekday boxes, eg [][x][][][x] for Tuesday Friday
// And the string that appears when you hover over it (Meets on Tuesday, Friday)

class WeekdayBoxes extends React.Component {
  render() {
    // Don't render anything if the class is online.
    if (this.props.section.online) {
      return null;
    }


    // Calculate the "Meets on Tuesday, Friday" or "No Meetings found" string that hovers over the weekday boxes
    const meetingDays = this.props.section.getWeekDaysAsStringArray();
    let meetingString;
    if (meetingDays.length === 0) {
      meetingString = 'No Meetings found';
    } else {
      meetingString = `Meets on ${meetingDays.join(', ')}`;
    }

    // Calculate the weekday boxes eg [][x][][][x] for Tuesday Friday
    let booleans = this.props.section.getWeekDaysAsBooleans();
    if (!this.props.section.meetsOnWeekends()) {
      booleans = booleans.slice(1, 6);
    }

    const booleanElements = booleans.map((meets, index) => {
      return (
        <div
          key={ index }
          className={ cx({
            weekDayBoxChecked: meets,
            weekDayBox: true,
          }) }
        />
      );
    });


    return (
      <div className={ `${css.inlineBlock} ${css.daysContainer}` } data-tip={ meetingString }>
        {booleanElements}
      </div>
    );
  }
}


WeekdayBoxes.propTypes = {
  section: PropTypes.object.isRequired,
};


export default WeekdayBoxes;
