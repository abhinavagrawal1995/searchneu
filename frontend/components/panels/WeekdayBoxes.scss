/*
 * This file is part of Search NEU and licensed under AGPL3. 
 * See the license file in the root folder for details. 
 */

.weekDayBox {
    width: 12px;
    height: 12px;
    display: inline-block;
    border: 2px solid #BBBBBB;
    border-radius: 3px;
    margin-right: 1px;
    margin-left: 1px;
}
.weekDayBoxChecked {
    background: #888;
}
.inlineBlock {
    display: inline-block;
}
.container {
    min-width: 70px;
}
