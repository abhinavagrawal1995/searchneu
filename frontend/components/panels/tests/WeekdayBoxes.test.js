/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */

import React from 'react';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import mockData from './mockData';
import WeekdayBoxes from '../WeekdayBoxes';

Enzyme.configure({ adapter: new Adapter() });


it('should render a section', () => {
  const result = shallow(<WeekdayBoxes section={ mockData.WMNS4520section } />);

  // Use Jest's snapshotting feature to ensure that the DOM does not change.
  // Jest saves these files in the __snapshots__ folder.
  expect(result).toMatchSnapshot();
});


it('should render another section', () => {
  const result = shallow(<WeekdayBoxes section={ mockData.cs0210.sections[0] } />);
  expect(result).toMatchSnapshot();
});

it('should not render a section if it\'s online', () => {
  const result = shallow(<WeekdayBoxes section={{ online: true }} />);

  expect(result.html()).toBe(null);
});
