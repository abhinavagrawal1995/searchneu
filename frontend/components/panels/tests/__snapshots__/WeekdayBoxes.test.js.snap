// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should behave... 1`] = `
<div
  className="inlineBlock daysContainer"
  data-tip="Meets on Tuesday, Friday"
>
  <div
    className="weekDayBox"
  />
  <div
    className="weekDayBoxChecked weekDayBox"
  />
  <div
    className="weekDayBox"
  />
  <div
    className="weekDayBox"
  />
  <div
    className="weekDayBoxChecked weekDayBox"
  />
</div>
`;
